tokio/src/runtime/task/core.rs

```diff
@@ -50,8 +50,9 @@ pub(super) struct Core<T: Future, S> {
     /// Either the future or the output.
     pub(super) stage: CoreStage<T>,
 
-    /// The task's ID, used for populating `JoinError`s.
-    pub(super) task_id: Id,
+    /// The tracing ID for this instrumented task.
+    #[cfg(all(tokio_unstable, feature = "tracing"))]
+    pub(super) id: Option<tracing::Id>,
```

Comment:
I'm a bit confused about what happened to the two id fields here. Were they swapped (one in header other in core)?

Reply:
Yes, originally this was done to potentially allow the `Header` size to fit into a cache line, though even the swap didn't fulfill that on 32-bit system. Now that the `owned` field was moved out of `Header` I believe there should be enough space to move the tracing id field back to `Header`, though since this field is only accessed in contexts in which we _can_ cast a `Header` back into `Core`, it probably makes sense to keep it there?!

Reply:
Actually, to clarify this was done since we need `task_id` in contexts in which we cannot cast `Header` back to `Cell` (at least without fairly large changes), so `task_id` has to be in `Header`. We moved `id` out of `Header` as an attempt to solve the cache line size problem.

Reply:
I guess it's fine.